core/dbt/clients/system.py

```diff
@@ -57,19 +59,30 @@ def find_matching(
     reobj = re.compile(regex, re.IGNORECASE)
 
     for relative_path_to_search in relative_paths_to_search:
+        # potential speedup for ignore_spec
+        # if ignore_spec.matches(relative_path_to_search):
+        #     continue
         absolute_path_to_search = os.path.join(root_path, relative_path_to_search)
         walk_results = os.walk(absolute_path_to_search)
```

Comment:
We should really avoid using `os.walk` or `os.path` in favor of `Pathlib`.  All the os.path stuff is old AF and has weird operating system specific gotchas.

Reply:
Agree! I think we probably leave it to the larger refactor? Same as the signatures for `read_files` calls are getting long. I also don't like that but feels like right now might be okay to leave it.